Representation/PaginatedRepresentation.php

```diff
@@ -109,12 +110,15 @@ public function hasNextPage()
 
     /**
      * @return int
+     * @throws \Exception
      */
     public function getNextPage()
     {
         if ($this->hasNextPage()) {
             return $this->page + 1;
         }
+
+        throw new \Exception('Next page does not exist');
```

Comment:
I am not sure that we should throw an exception, if there is no new page, the result should be empty. If we throw an exception, then we should handle response always with try/catch

Reply:
I've added it due to returning type.
If you hint that integer will be returned, then exception should be thrown if nothing to return.
Retunring zero/one as default - sounds bad to me.
We could actually hint int|null - but I don't like it too :)

@RuslanZavacky What will you recommend to return here?

Reply:
I'd suggest checking the context where it is used:

```php
'next' => $representation->hasNextPage() ? $this->getUriForPage($representation->getNextPage()) : null,
```
It is fine here to just return number or null. As  `hasNextPage` page check, checks for:

```php
return $this->page < $this->getPages();
```

Reply:
Deal. Done.